### DISCUSSION TITLE
Clarification on Codespaces with Docker Compose Files

### DISCUSSION THREAD

**Discussion by metalli-tony** [community/community#44701] [Codespaces] [answered]
### Select Topic Area

Question

### Body

I'm new to Codepsaces and somewhat new to Docker. In going through the documentation, I've always been under the impression that a Codespace is a single container running on a VM. I recently came across the documentation for using a Docker Compose file with Codespaces. My questions is fairly simple: When using a Docker Compose file with Codespaces and defining several services (db, cache, messaging, webapp, etc) within the Docker Compose file, does that create multiple Docker containers, or multiple services within the same container? Appreciate any help. Thanks!

**Comment by 8Gitbrix** [comment, marked as answer]
It should work as docker normally functions, meaning docker compose will create multiple docker containers.
When you open codespace, the container that you connect to is the main container.
